File: src/main/java/hyeon9mak/multidatasourcequerycounter/QueryCounterLoggingLevelProperties.java
package hyeon9mak.multidatasourcequerycounter;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@ConfigurationProperties(prefix = "query-counter.logging.level")
@Component
public class QueryCounterLoggingLevelProperties {

    private final QueryCounterLoggingLevelProperty info;
    private final QueryCounterLoggingLevelProperty warn;
    private final QueryCounterLoggingLevelProperty error;

    public QueryCounterLoggingLevelProperties() {
        this.info = new QueryCounterLoggingLevelProperty();
        this.warn = new QueryCounterLoggingLevelProperty();
        this.error = new QueryCounterLoggingLevelProperty();
    }

    public QueryCounterLoggingLevelProperty getInfo() {
        return info;
    }

    public QueryCounterLoggingLevelProperty getWarn() {
        return warn;
    }

    public QueryCounterLoggingLevelProperty getError() {
        return error;
    }
}

File: src/main/kotlin/hyeon9mak/multidatasourcequerycounter/QueryCounterLoggingLevelProperties.kt
package hyeon9mak.multidatasourcequerycounter

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@ConfigurationProperties(prefix = "query-counter.logging.level")
@Component
data class QueryCounterLoggingLevelProperties(
  val info: QueryCounterLoggingLevelProperty = QueryCounterLoggingLevelProperty(),
  val warn: QueryCounterLoggingLevelProperty = QueryCounterLoggingLevelProperty(),
  val error: QueryCounterLoggingLevelProperty = QueryCounterLoggingLevelProperty(),
) {
  data class QueryCounterLoggingLevelProperty(
    var enable: Boolean = false,
    var count: Int = 1
  )
}
